lib/util.js

```diff
@@ -0,0 +1,170 @@
+  }
+}
+
+function consumesFormOnly (schema) {
+  const consumes = schema.consumes
+  return (
+    consumes &&
+      consumes.length === 1 &&
+      (consumes[0] === 'application/x-www-form-urlencoded' ||
+        consumes[0] === 'multipart/form-data')
+  )
+}
+
+// For supported keys read:
+// https://swagger.io/docs/specification/2-0/describing-parameters/
+function plainJsonObjectToSwagger2 (container, jsonSchema, externalSchemas) {
+  const obj = localRefResolve(jsonSchema, externalSchemas)
+  let toSwaggerProp
+  switch (container) {
+    case 'query':
+      toSwaggerProp = function (properyName, jsonSchemaElement) {
+        jsonSchemaElement.in = container
+        jsonSchemaElement.name = properyName
+        return jsonSchemaElement
+      }
+      break
+    case 'formData':
+      toSwaggerProp = function (properyName, jsonSchemaElement) {
+        delete jsonSchemaElement.$id
+        jsonSchemaElement.in = container
+        jsonSchemaElement.name = properyName
+
+        // https://json-schema.org/understanding-json-schema/reference/non_json_data.html#contentencoding
+        if (jsonSchemaElement.contentEncoding === 'binary') {
+          delete jsonSchemaElement.contentEncoding // Must be removed
+          jsonSchemaElement.type = 'file'
+        }
+
+        return jsonSchemaElement
+      }
+      break
+    case 'path':
+      toSwaggerProp = function (properyName, jsonSchemaElement) {
+        jsonSchemaElement.in = container
+        jsonSchemaElement.name = properyName
+        jsonSchemaElement.required = true
+        return jsonSchemaElement
+      }
+      break
+    case 'header':
+      toSwaggerProp = function (properyName, jsonSchemaElement) {
+        return {
+          in: 'header',
+          name: properyName,
+          required: jsonSchemaElement.required,
+          description: jsonSchemaElement.description,
+          type: jsonSchemaElement.type
+        }
+      }
+      break
+  }
+
+  return Object.keys(obj).reduce((acc, propKey) => {
+    acc.push(toSwaggerProp(propKey, obj[propKey]))
+    return acc
+  }, [])
+}
+
+function swagger2ParametersToOpenapi3 (jsonSchema) {
+  jsonSchema.schema = {}
+  jsonSchema.schema.type = jsonSchema.type
+  if (jsonSchema.type === 'object') {
+    jsonSchema.schema.properties = jsonSchema.properties
+  }
+  if (jsonSchema.type === 'array') {
+    jsonSchema.schema.items = jsonSchema.items
+  }
+  delete jsonSchema.type
+  delete jsonSchema.properties
+  delete jsonSchema.items
+  return jsonSchema
+}
+
+function localRefResolve (jsonSchema, externalSchemas) {
+  if (jsonSchema.type && jsonSchema.properties) {
+    // for the shorthand querystring/params/headers declaration
+    const propertiesMap = Object.keys(jsonSchema.properties).reduce((acc, h) => {
+      const required = (jsonSchema.required && jsonSchema.required.indexOf(h) >= 0) || false
+      const newProps = Object.assign({}, jsonSchema.properties[h], { required })
```

Comment:
Why use `Object.assign` instead of a simple spread?

Reply:
Most of the functions in ```util.js``` is extracted from the ```dynamic.js```. I do not really know what is the actual behavior or what is the goal of most of the functions.
For the ```Object.assign```, maybe it want to preserve the ```Symbol``` property.